paasta_tools/marathon_tools.py

```diff
@@ -368,6 +370,28 @@ def format_marathon_app_dict(self):
 
         net = get_mesos_network_for_net(self.get_net())
 
+        try:
+            constraints = self.get_calculated_constraints(
+                system_paasta_config=system_paasta_config,
+                service_namespace_config=service_namespace_config
+            )
+        except NoSlavesAvailableError as e:
+            cluster = system_paasta_config.get_cluster()
+            error_msg = (
+                "There are no PaaSTA slaves that can run %s in cluster %s\n" % (self.instance, cluster) +
+                "Double check the cluster and the configured constraints/pool/whitelist.\n"
+                "Error was: %s" % str(e)
+            )
+            monitoring_tools.send_event(
+                service=self.service,
+                check_name="PaaSTA could not meet your specified constraints!",
+                overrides={},
+                status=pysensu_yelp.Status.CRITICAL,
+                output=error_msg,
+                soa_dir=self.soa_dir,
+            )
+            constraints = []
```

Comment:
This is too dangerous. If we can't calculate the constraints we need to bail. But this may be too deep to bail, I think you'll have to catch this further up and return a non-0 code.

The most important part here is that we don't crash deployd/smj entirely.